exporters/dsexporter/dsexporter.go

```diff
@@ -1,39 +1,109 @@
 package main
 
 import (
+	"context"
+	"encoding/json"
 	"fmt"
 	"net/http"
+	"os"
+	"strings"
 
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"k8s.io/client-go/kubernetes"
+	// "k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
 )
 
+const service = "https://console-openshift-console.apps.stone-prd-rh01.pg1f.p1.openshiftapps.com/k8s/ns/appstudio-grafana/services"
+const check = "prometheus-appstudio-ds"
+var getInClusterConfig = config.GetConfigOrDie
+var getNewForConfig = kubernetes.NewForConfigOrDie
+
 type CustomCollector struct {
-	requestCounter prometheus.Counter
+	konfluxDSExporter *prometheus.GaugeVec
 }
 
 // Creating a new instance of CustomCollector.
 func NewCustomCollector() *CustomCollector {
 	return &CustomCollector{
-		requestCounter: prometheus.NewCounter(prometheus.CounterOpts{
-			Name: "request_count",
-			Help: "Number of requests handled by the handler",
-		}),
+		konfluxDSExporter: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "konflux_dsexporter",
+			Help: "Availability of konflux service/component",
```

Comment:
The help is for the specific metric, I think it should be more informative

```suggestion
			Help: "Availability of prometheus-appstudio-ds Grafana datasource",
```

Reply:
Or if we don't want to expose `appstudio` unecessarily (since it's an old name):

```suggestion
			Help: "Availability of the Konflux default grafana datasource",
```